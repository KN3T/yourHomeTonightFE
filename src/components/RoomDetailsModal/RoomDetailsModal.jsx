/* eslint-disable react/prop-types */
import { Button, Card, Col, Image, Modal, Row, message } from 'antd';
import React, { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { AiOutlineCheck } from 'react-icons/ai';
import { BsFillPeopleFill } from 'react-icons/bs';
import { IoBedSharp } from 'react-icons/io5';
import { MdOutlineChildCare } from 'react-icons/md';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';

import { addOrder } from '../../store/Slice/Booking/BookingSlice';
import formatCurrency from '../../utils/formatCurrency';
import './RoomDetailsModal.scss';

const RoomDetailsModal = (props) => {
  const {
    isModalVisible,
    handleOk,
    handleCancel,
    roomData,
    roomImages,
    dataOrder,
    setIsModalVisible,
  } = props;

  const navigate = useNavigate();
  const [bigImage, setBigImage] = useState(roomImages[0]);
  const [orderData, setOrderData] = useState(dataOrder);
  const dispach = useDispatch();

  useEffect(() => {
    setBigImage(roomImages[0]);
  }, [roomData]);

  useEffect(() => {
    setOrderData(dataOrder);
  }, [dataOrder]);

  const { t, i18n } = useTranslation();
  const currentLanguage = i18n.language;

  const handleBooking = () => {
    if (localStorage.getItem('userData')) {
      dispach(addOrder(orderData));
      window.localStorage.setItem('bookingData', JSON.stringify(dataOrder));
      navigate('/checkout');
    } else {
      navigate('/login');
    }
  };
  return (
    <Modal
      visible={isModalVisible}
      onOk={handleOk}
      onCancel={handleCancel}
      className="room__details__container"
      footer={false}
    >
      <Card bordered={false}>
        <Row gutter={[16, 0]}>
          <Col span={14} className="images__section">
            <div className="big__images__wrapper">
              <Image
                className="big__images__content"
                src={bigImage.src}
                fallback="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAMIAAADDCAYAAADQvc6UAAABRWlDQ1BJQ0MgUHJvZmlsZQAAKJFjYGASSSwoyGFhYGDIzSspCnJ3UoiIjFJgf8LAwSDCIMogwMCcmFxc4BgQ4ANUwgCjUcG3awyMIPqyLsis7PPOq3QdDFcvjV3jOD1boQVTPQrgSkktTgbSf4A4LbmgqISBgTEFyFYuLykAsTuAbJEioKOA7DkgdjqEvQHEToKwj4DVhAQ5A9k3gGyB5IxEoBmML4BsnSQk8XQkNtReEOBxcfXxUQg1Mjc0dyHgXNJBSWpFCYh2zi+oLMpMzyhRcASGUqqCZ16yno6CkYGRAQMDKMwhqj/fAIcloxgHQqxAjIHBEugw5sUIsSQpBobtQPdLciLEVJYzMPBHMDBsayhILEqEO4DxG0txmrERhM29nYGBddr//5/DGRjYNRkY/l7////39v///y4Dmn+LgeHANwDrkl1AuO+pmgAAADhlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAAqACAAQAAAABAAAAwqADAAQAAAABAAAAwwAAAAD9b/HnAAAHlklEQVR4Ae3dP3PTWBSGcbGzM6GCKqlIBRV0dHRJFarQ0eUT8LH4BnRU0NHR0UEFVdIlFRV7TzRksomPY8uykTk/zewQfKw/9znv4yvJynLv4uLiV2dBoDiBf4qP3/ARuCRABEFAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghgg0Aj8i0JO4OzsrPv69Wv+hi2qPHr0qNvf39+iI97soRIh4f3z58/u7du3SXX7Xt7Z2enevHmzfQe+oSN2apSAPj09TSrb+XKI/f379+08+A0cNRE2ANkupk+ACNPvkSPcAAEibACyXUyfABGm3yNHuAECRNgAZLuYPgEirKlHu7u7XdyytGwHAd8jjNyng4OD7vnz51dbPT8/7z58+NB9+/bt6jU/TI+AGWHEnrx48eJ/EsSmHzx40L18+fLyzxF3ZVMjEyDCiEDjMYZZS5wiPXnyZFbJaxMhQIQRGzHvWR7XCyOCXsOmiDAi1HmPMMQjDpbpEiDCiL358eNHurW/5SnWdIBbXiDCiA38/Pnzrce2YyZ4//59F3ePLNMl4PbpiL2J0L979+7yDtHDhw8vtzzvdGnEXdvUigSIsCLAWavHp/+qM0BcXMd/q25n1vF57TYBp0a3mUzilePj4+7k5KSLb6gt6ydAhPUzXnoPR0dHl79WGTNCfBnn1uvSCJdegQhLI1vvCk+fPu2ePXt2tZOYEV6/fn31dz+shwAR1sP1cqvLntbEN9MxA9xcYjsxS1jWR4AIa2Ibzx0tc44fYX/16lV6NDFLXH+YL32jwiACRBiEbf5KcXoTIsQSpzXx4N28Ja4BQoK7rgXiydbHjx/P25TaQAJEGAguWy0+2Q8PD6/Ki4R8EVl+bzBOnZY95fq9rj9zAkTI2SxdidBHqG9+skdw43borCXO/ZcJdraPWdv22uIEiLA4q7nvvCug8WTqzQveOH26fodo7g6uFe/a17W3+nFBAkRYENRdb1vkkz1CH9cPsVy/jrhr27PqMYvENYNlHAIesRiBYwRy0V+8iXP8+/fvX11Mr7L7ECueb/r48eMqm7FuI2BGWDEG8cm+7G3NEOfmdcTQw4h9/55lhm7DekRYKQPZF2ArbXTAyu4kDYB2YxUzwg0gi/41ztHnfQG26HbGel/crVrm7tNY+/1btkOEAZ2M05r4FB7r9GbAIdxaZYrHdOsgJ/wCEQY0J74TmOKnbxxT9n3FgGGWWsVdowHtjt9Nnvf7yQM2aZU/TIAIAxrw6dOnAWtZZcoEnBpNuTuObWMEiLAx1HY0ZQJEmHJ3HNvGCBBhY6jtaMoEiJB0Z29vL6ls58vxPcO8/zfrdo5qvKO+d3Fx8Wu8zf1dW4p/cPzLly/dtv9Ts/EbcvGAHhHyfBIhZ6NSiIBTo0LNNtScABFyNiqFCBChULMNNSdAhJyNSiECRCjUbEPNCRAhZ6NSiAARCjXbUHMCRMjZqBQiQIRCzTbUnAARcjYqhQgQoVCzDTUnQIScjUohAkQo1GxDzQkQIWejUogAEQo121BzAkTI2agUIkCEQs021JwAEXI2KoUIEKFQsw01J0CEnI1KIQJEKNRsQ80JECFno1KIABEKNdtQcwJEyNmoFCJAhELNNtScABFyNiqFCBChULMNNSdAhJyNSiECRCjUbEPNCRAhZ6NSiAARCjXbUHMCRMjZqBQiQIRCzTbUnAARcjYqhQgQoVCzDTUnQIScjUohAkQo1GxDzQkQIWejUogAEQo121BzAkTI2agUIkCEQs021JwAEXI2KoUIEKFQsw01J0CEnI1KIQJEKNRsQ80JECFno1KIABEKNdtQcwJEyNmoFCJAhELNNtScABFyNiqFCBChULMNNSdAhJyNSiECRCjUbEPNCRAhZ6NSiAARCjXbUHMCRMjZqBQiQIRCzTbUnAARcjYqhQgQoVCzDTUnQIScjUohAkQo1GxDzQkQIWejUogAEQo121BzAkTI2agUIkCEQs021JwAEXI2KoUIEKFQsw01J0CEnI1KIQJEKNRsQ80JECFno1KIABEKNdtQcwJEyNmoFCJAhELNNtScABFyNiqFCBChULMNNSdAhJyNSiEC/wGgKKC4YMA4TAAAAABJRU5ErkJggg=="
              />
            </div>
            <div className="images__group">
              {roomData.images.map((item, index) => {
                return (
                  <Image
                    preview={false}
                    src={item.src}
                    key={index}
                    onClick={() => setBigImage(item)}
                  />
                );
              })}
            </div>
          </Col>
          <Col className="info__section" span={10}>
            <h2 className="info__name">{roomData.type}</h2>
            <div className="info__description">
              <h2 className="info__title">{t('room.description')}</h2>
              <ul>
                <li className="description__item">
                  <IoBedSharp /> {t('room.beds')}: {roomData.beds}
                </li>
                <li className="description__item">
                  <BsFillPeopleFill /> {t('room.adults')}: {roomData.adults}
                </li>
                <li className="description__item">
                  <MdOutlineChildCare /> {t('room.children')}:{' '}
                  {roomData.children}
                </li>
                <li className="description__item">{roomData.description}</li>
              </ul>
            </div>
            <div className="info__facilities">
              <h2 className="info__title">{t('room.facilities')}</h2>
              <ul className="facilities__list">
                {roomData.asset.map((item, index) => {
                  return (
                    <li key={index}>
                      <span>
                        <AiOutlineCheck />
                        {item}
                      </span>
                    </li>
                  );
                })}
              </ul>
            </div>
          </Col>
        </Row>
        <Row gutter={[[0, 0]]} className="bottom__section">
          <Col span={24} className="bottom__wrapper">
            <div className="room__price">
              <strong>
                {t('room.only')}{' '}
                {t('room.price_value', {
                  val: formatCurrency(roomData.price, currentLanguage),
                })}{' '}
                {t('room.per_night')}
              </strong>
            </div>
            <div className="booking__button">
              {dataOrder.checkIn && dataOrder.checkOut ? (
                <Button
                  onClick={() => {
                    handleBooking();
                    setIsModalVisible(false);
                  }}
                  type="primary"
                >
                  {t('room.reverse_now')}
                </Button>
              ) : (
                <Button
                  onClick={() =>
                    message.error(
                      'You must select date check in, check out first, please!'
                    )
                  }
                  type="primary"
                >
                  {t('room.reverse_now')}
                </Button>
              )}
            </div>
          </Col>
        </Row>
      </Card>
    </Modal>
  );
};

export default RoomDetailsModal;
