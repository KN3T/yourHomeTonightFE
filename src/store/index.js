import { configureStore } from '@reduxjs/toolkit';

import HotelsSlice from './Slice/Hotels/HotelsSlice';
import RoomsSlice from './Slice/Rooms/RoomsSlice'
import BookingSlice from './Slice/Booking/BookingSlice'

export const store = configureStore({
  reducer: {
    hotels: HotelsSlice,
    rooms: RoomsSlice,
    booking: BookingSlice
  },
});
