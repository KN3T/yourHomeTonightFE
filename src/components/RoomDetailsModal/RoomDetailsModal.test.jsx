import { render, screen } from '@testing-library/react';
import moment from 'moment';
import React from 'react';
import { Provider } from 'react-redux';
import { HashRouter } from 'react-router-dom';

import { store } from '../../store';
import RoomDetailsModal from './RoomDetailsModal';

describe('Room details modal', () => {
  it('Should return a modal show details information about room', () => {
    const roomData = {
      id: 15,
      number: 1,
      type: 'Standard Double Room',
      beds: 2,
      price: 25,
      adults: 4,
      children: 2,
      description:
        'This double room has a cable TV, air conditioning and seating area.',
      images: [{ imagesId: 1, src: 'imagesURL' }],
      asset: ['abc', 'xyz'],
      rating: 5,
    };
    const dataOrder = {
      checkIn: parseInt(moment(1657346717).format('X')),
      checkOut: parseInt(moment(1657493084).format('X')),
      selectedRoom: roomData,
    };
    const roomDetailsModal = render(
      <Provider store={store}>
        <HashRouter>
          <RoomDetailsModal
            isModalVisible="true"
            handleBooking={() => console.log('Navigate to checkout page')}
            handleCancel={() => console.log('Turn off modal')}
            roomData={roomData}
            roomImages={'Images URL'}
            dataOrder={dataOrder}
            setIsModalVisible={() => console.log(123)}
          />
        </HashRouter>
      </Provider>
    );
    expect(roomDetailsModal).toMatchSnapshot();
  });

  it('should render button booking', () => {
    const btn = screen.queryByTestId('btn__booking');
    expect(
      <btn
        handleBooking={() => console.log('booking button')}
        setIsModalVisible={() => console.log('close modal')}
      />
    ).toMatchSnapshot();
  });
});
