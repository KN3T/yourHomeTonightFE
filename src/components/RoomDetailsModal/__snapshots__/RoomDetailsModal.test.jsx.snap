// Vitest Snapshot v1

exports[`Room details modal > Should return a modal show details information about room 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="ant-scrolling-effect"
    style="overflow: hidden; overflow-x: hidden; overflow-y: hidden;"
  >
    <div />
    <div>
      <div
        class="ant-modal-root"
      >
        <div
          class="ant-modal-mask"
        />
        <div
          class="ant-modal-wrap"
          tabindex="-1"
        >
          <div
            aria-modal="true"
            class="ant-modal room__details__container"
            role="dialog"
            style="width: 520px;"
          >
            <div
              aria-hidden="true"
              style="width: 0px; height: 0px; overflow: hidden; outline: none;"
              tabindex="0"
            />
            <div
              class="ant-modal-content"
            >
              <button
                aria-label="Close"
                class="ant-modal-close"
                type="button"
              >
                <span
                  class="ant-modal-close-x"
                >
                  <span
                    aria-label="close"
                    class="anticon anticon-close ant-modal-close-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
              <div
                class="ant-modal-body"
              >
                <div
                  class="ant-card"
                >
                  <div
                    class="ant-card-body"
                  >
                    <div
                      class="ant-row"
                      style="margin-left: -8px; margin-right: -8px;"
                    >
                      <div
                        class="ant-col ant-col-14 images__section"
                        style="padding-left: 8px; padding-right: 8px;"
                      >
                        <div
                          class="big__images__wrapper"
                        >
                          <div
                            class="ant-image"
                          >
                            <img
                              class="ant-image-img big__images__content"
                            />
                            <div
                              class="ant-image-mask"
                            >
                              <div
                                class="ant-image-mask-info"
                              >
                                <span
                                  aria-label="eye"
                                  class="anticon anticon-eye"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="eye"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                                    />
                                  </svg>
                                </span>
                                Preview
                              </div>
                            </div>
                          </div>
                        </div>
                        <div
                          class="images__group"
                        >
                          <div
                            class="ant-image"
                          >
                            <img
                              class="ant-image-img"
                              src="imagesURL"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class="ant-col ant-col-10 info__section"
                        style="padding-left: 8px; padding-right: 8px;"
                      >
                        <h2
                          class="info__name"
                        >
                          Standard Double Room
                        </h2>
                        <div
                          class="room__price"
                        >
                          <h3>
                            room.only
                             
                            <span
                              style="font-weight: 700;"
                            >
                              room.price_value
                               
                            </span>
                            room.per_night
                          </h3>
                        </div>
                        <div
                          class="info__description"
                        >
                          <h2
                            class="info__title"
                          >
                            room.description
                          </h2>
                          <ul>
                            <li
                              class="description__item"
                            >
                              <svg
                                fill="currentColor"
                                height="1em"
                                stroke="currentColor"
                                stroke-width="0"
                                viewBox="0 0 512 512"
                                width="1em"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M432 224V96a16 16 0 00-16-16H96a16 16 0 00-16 16v128a48 48 0 00-48 48v160h36v-32h376v32h36V272a48 48 0 00-48-48zm-192 0H120v-32a16 16 0 0116-16h88a16 16 0 0116 16zm32-32a16 16 0 0116-16h88a16 16 0 0116 16v32H272z"
                                />
                              </svg>
                               
                              room.beds
                              : 
                              2
                            </li>
                            <li
                              class="description__item"
                            >
                              <svg
                                fill="currentColor"
                                height="1em"
                                stroke="currentColor"
                                stroke-width="0"
                                viewBox="0 0 16 16"
                                width="1em"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M7 14s-1 0-1-1 1-4 5-4 5 3 5 4-1 1-1 1H7zm4-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
                                />
                                <path
                                  d="M5.216 14A2.238 2.238 0 0 1 5 13c0-1.355.68-2.75 1.936-3.72A6.325 6.325 0 0 0 5 9c-4 0-5 3-5 4s1 1 1 1h4.216z"
                                  fill-rule="evenodd"
                                />
                                <path
                                  d="M4.5 8a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5z"
                                />
                              </svg>
                               
                              room.adults
                              : 
                              4
                            </li>
                            <li
                              class="description__item"
                            >
                              <svg
                                fill="currentColor"
                                height="1em"
                                stroke="currentColor"
                                stroke-width="0"
                                viewBox="0 0 24 24"
                                width="1em"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M0 0h24v24H0V0z"
                                  fill="none"
                                />
                                <circle
                                  cx="14.5"
                                  cy="10.5"
                                  r="1.25"
                                />
                                <circle
                                  cx="9.5"
                                  cy="10.5"
                                  r="1.25"
                                />
                                <path
                                  d="M22.94 11.34a4.008 4.008 0 00-2.81-3.17 9.114 9.114 0 00-2.19-2.91C16.36 3.85 14.28 3 12 3s-4.36.85-5.94 2.26c-.92.81-1.67 1.8-2.19 2.91a3.994 3.994 0 00-2.81 3.17c-.04.21-.06.43-.06.66 0 .23.02.45.06.66a4.008 4.008 0 002.81 3.17 8.977 8.977 0 002.17 2.89C7.62 20.14 9.71 21 12 21s4.38-.86 5.97-2.28c.9-.8 1.65-1.79 2.17-2.89a3.998 3.998 0 002.8-3.17c.04-.21.06-.43.06-.66 0-.23-.02-.45-.06-.66zM19 14c-.1 0-.19-.02-.29-.03-.2.67-.49 1.29-.86 1.86C16.6 17.74 14.45 19 12 19s-4.6-1.26-5.85-3.17c-.37-.57-.66-1.19-.86-1.86-.1.01-.19.03-.29.03-1.1 0-2-.9-2-2s.9-2 2-2c.1 0 .19.02.29.03.2-.67.49-1.29.86-1.86C7.4 6.26 9.55 5 12 5s4.6 1.26 5.85 3.17c.37.57.66 1.19.86 1.86.1-.01.19-.03.29-.03 1.1 0 2 .9 2 2s-.9 2-2 2zm-7 3c2.01 0 3.74-1.23 4.5-3h-9c.76 1.77 2.49 3 4.5 3z"
                                />
                              </svg>
                               
                              room.children
                              :
                               
                              2
                            </li>
                            <li
                              class="description__item"
                            >
                              This double room has a cable TV, air conditioning and seating area.
                            </li>
                          </ul>
                        </div>
                        <div
                          class="info__facilities"
                        >
                          <h2
                            class="info__title"
                          >
                            room.facilities
                          </h2>
                          <ul
                            class="facilities__list"
                          >
                            <li>
                              <span>
                                <svg
                                  fill="currentColor"
                                  height="1em"
                                  stroke="currentColor"
                                  stroke-width="0"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
                                  />
                                </svg>
                                abc
                              </span>
                            </li>
                            <li>
                              <span>
                                <svg
                                  fill="currentColor"
                                  height="1em"
                                  stroke="currentColor"
                                  stroke-width="0"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
                                  />
                                </svg>
                                xyz
                              </span>
                            </li>
                          </ul>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-row bottom__section"
                    >
                      <div
                        class="ant-col ant-col-24 bottom__wrapper"
                      >
                        <div
                          class="booking__button"
                        >
                          <button
                            class="ant-btn ant-btn-primary"
                            type="button"
                          >
                            <span>
                              room.reverse_now
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              style="width: 0px; height: 0px; overflow: hidden; outline: none;"
              tabindex="0"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
